src/utils/are_codecs_compatible.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * This function is a shortcut that helps differentiate two codecs.
+ * 
+ * @param codecA 
+ * @param codecB 
+ * @returns A boolean that tell whether or not those two codecs provided are even.
+ */
+function areCodecsCompatible(codecA: string, codecB: string) {
+    return codecA === codecB;
```

Comment:
We should ignore things like the profile part if it exists.

Reply:
Not sure to understood what u mean by `profile part`?

Reply:
Well perhaps it's not the right term but I'm speaking of the latter part of the codec string, like in https://wiki.whatwg.org/wiki/Video_type_parameters#Audio_Codecs_3.

Here both:
  - Low-Complexity AAC: mp4a.40.2
  - High-Efficiency AAC: mp4a.40.5

Should be compatible between one another despite having different codec strings.

This is why for example the shaka-player compares like this:
```js
  static areStreamsCompatible_(s0, s1) {
    // Basic mime types and basic codecs need to match.
    // For example, we can't adapt between WebM and MP4,
    // nor can we adapt between mp4a.* to ec-3.
    // We can switch between text types on the fly,
    // so don't run this check on text.
    if (s0.mimeType != s1.mimeType) {
      return false;
    }

    if (s0.codecs.split('.')[0] != s1.codecs.split('.')[0]) {
      return false;
    }

    return true;
  }
```

Reply:
Note that technically, we can have a `.` in the mimetype string. So we might want to split-up that part to (until the first `;` perhaps?)

Reply:
Ok, yes I was thinking to this.

yea, Im gonna do what I've previously done about that comparison. Thanks!

Reply:
Not to forget that there could be multiple mimetype parameters here (not only `codecs=`).
We might want to detect that also (it seems that each parameters is separated by a `;` but I'm not sure)

Reply:
I found this: https://tools.ietf.org/html/rfc2045#section-5.1

It is defined for the content-type header, but it seems to work the same way than the mimetype string of a SourceBuffer here.

Reply:
You mean that it could have multiple `mimeType`?

For me, this is the format: `"video/mp4;codecs=\"avc1.4d401e\""` ?

Reply:
No multiple parameters like the "codecs" one.

Like "audio/mp4;toto=5;codecs=mp4a.40.2;test=4"

Reply:
We could do something like:
```js
function areCodecsCompatible(a, b) {
  const splittedA = a.split(";");
  const splittedB = b.split(";");

  const mimeTypeA = splittedA[0];
  const mimeTypeB = splittedB[0];

  if (mimeTypeA !== mimeTypeB) {
    return false;
  }

  const codecA = getCodecParameter()
  for (let i = 1; i < splittedA.length; i++) {
    const trimmedA = splittedA[i].trim();
    if (startsWith(trimmedA, "codecs=")) {
      const codecStringA = trimmedA.substring(7);
      const splittedCodecA = codecStringA.split(".");
      const codecA = splittedCodecA.length > 0 ? splittedCodecA[0] : 
                                                                             "";

       // ...
       // ending with something like:
       // return codecA === codecB;
    }
  }
}

Reply:
Ho understood!

I tried "my" version, what do you think?

Open to idea :)